stdlib/REPL/src/docview.jl

```diff
@@ -166,7 +166,7 @@ struct Logged{F}
     collection::Set{Pair{Module,Symbol}}
 end
 function (la::Logged)(m::Module, s::Symbol)
-    m !== la.mod && !Base.ispublic(m, s) && push!(la.collection, m => s)
+    m !== la.mod && Base.isdefined(m, s) && !Base.ispublic(m, s) && push!(la.collection, m => s)
```

Comment:
I am not quite sure what the intent is with this code, but checking whether a value is defined seems like a code smell when everything else here is checking for properties of the binding.

Reply:
I want to log any instances of "this global access will succeed but is not public". If `isdefined` is false, then we'll get an error (or an explicit note that the access is not defined) and there is no need for a warning. I think that `Base.isdefined(m, s) && !Base.ispublic(m, s)` is equivalent to the (not defined) `Base.isprivate(m, s)`.